src/System.Linq.Expressions/src/System/Dynamic/Utils/DelegateHelpers.cs

```diff
@@ -42,7 +42,7 @@ internal static Delegate CreateObjectArrayDelegate(Type delegateType, Func<objec
         // return (TRet)ret;
         private static Delegate CreateObjectArrayDelegateRefEmit(Type delegateType, Func<object[], object> handler)
         {
-            MethodInfo delegateInvokeMethod = delegateType.GetMethod("Invoke");
+            MethodInfo delegateInvokeMethod = delegateType.GetMethod("Invoke", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
Now, that it is more than trivial GetMethoid, it should be moved into some shared helper. And it woudl be nice to assert that it is only called on delegate types.